logprep/processor/concatenator/rule.py

```diff
@@ -1,151 +1,126 @@
-    ]
+    @define(kw_only=True)
+    class Config:
+        """RuleConfig for Concatenator"""
 
-    def __init__(self, filter_rule: FilterExpression, rule_config: dict):
+        source_fields: list = field(
+            validator=validators.deep_iterable(
+                member_validator=validators.instance_of(str),
+                iterable_validator=validators.instance_of(list),
+            )
+        )
+        """The source fields that should be concatenated, can contain dotted field paths."""
+        target_field: str = field(validator=validators.instance_of(str))
+        """The field in which the result should be written to."""
+        seperator: str = field(validator=validators.instance_of(str))
+        """The character(s) that should be used between the combined source field values."""
+        overwrite_target: bool = field(validator=validators.instance_of(bool))
+        """Defines whether the target field should be overwritten if it exists already."""
+        delete_source_fields: bool = field(validator=validators.instance_of(bool))
+        """Defines whether the source fields should be deleted after they have been combined
+        to the new field."""
+
+    _config: "ConcatenatorRule.Config"
+
+    def __init__(self, filter_rule: FilterExpression, config: "ConcatenatorRule.Config"):
         """
         Instantiate ConcatenatorRule based on a given filter and processor configuration.
 
         Parameters
         ----------
         filter_rule : FilterExpression
             Given lucene filter expression as a representation of the rule's logic.
-        rule_config: dict
+        config : "ConcatenatorRule.Config"
             Configuration fields from a given pipeline that refer to the processor instance.
         """
         super().__init__(filter_rule)
-        self._config = rule_config
-        self._source_fields = self._config.get("source_fields")
-        self._target_field = self._config.get("target_field")
-        self._seperator = self._config.get("seperator")
-        self._overwrite_target = self._config.get("overwrite_target")
-        self._delete_source_fields = self._config.get("delete_source_fields")
+        self._config = config
 
     def __eq__(self, other: "ConcatenatorRule") -> bool:
         return all(
             [
                 other.filter == self._filter,
-                self._source_fields == other.source_fields,
-                self._target_field == other.target_field,
-                self._seperator == other.seperator,
-                self._overwrite_target == other.overwrite_target,
-                self._delete_source_fields == other.delete_source_fields,
+                self.source_fields == other.source_fields,
+                self.target_field == other.target_field,
+                self.seperator == other.seperator,
+                self.overwrite_target == other.overwrite_target,
+                self.delete_source_fields == other.delete_source_fields,
```

Comment:
```suggestion
                self._config == other._config
```
easy? ;-)

Reply:
haha true :smile: I changed it to your solution.